cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala

```diff
@@ -300,6 +300,165 @@ class MultivariateAnomalyEstimator(override val uid: String) extends Estimator[D
+  }
+
+  protected def getInternalTransformer(schema: StructType): PipelineModel = {
+    val dynamicParamColName = DatasetExtensions.findUnusedColumnName("dynamic", schema)
+    val badColumns = getVectorParamMap.values.toSet.diff(schema.fieldNames.toSet)
+    assert(badColumns.isEmpty,
+      s"Could not find dynamic columns: $badColumns in columns: ${schema.fieldNames.toSet}")
+
+    val missingRequiredParams = this.getRequiredParams.filter {
+      p => this.get(p).isEmpty && this.getDefault(p).isEmpty
+    }
+    assert(missingRequiredParams.isEmpty,
+      s"Missing required params: ${missingRequiredParams.map(s => s.name).mkString("(", ", ", ")")}")
+
+    val dynamicParamCols = getVectorParamMap.values.toList.map(col) match {
+      case Nil => Seq(lit(false).alias("placeholder"))
+      case l => l
+    }
+
+    val blobHelper = if (this.get(connectionString).nonEmpty) {
+      AnomalyDetectionBlobHelpers(getConnectionString, getContainerName, getTimestampCol)
+    } else {
+      AnomalyDetectionBlobHelpers(getStorageName, getStorageKey, getEndpoint,
+        getSASToken, getContainerName, getTimestampCol)
+    }
+    this.setSource(blobHelper.upload(getSourceDF))
+    blobName = blobHelper.blobName
+
+    val stages = Array(
+      Lambda(_.withColumn(dynamicParamColName, struct(dynamicParamCols: _*))),
+      new SimpleHTTPTransformer()
+        .setInputCol(dynamicParamColName)
+        .setOutputCol(getOutputCol)
+        .setInputParser(getInternalInputParser(schema))
+        .setOutputParser(getInternalOutputParser(schema))
+        .setHandler(handlingFunc)
+        .setConcurrency(getConcurrency)
+        .setConcurrentTimeout(get(concurrentTimeout))
+        .setErrorCol(getErrorCol),
+      new DropColumns().setCol(dynamicParamColName)
+    )
+
+    NamespaceInjections.pipelineModel(stages)
+  }
+
+  override def fit(dataset: Dataset[_]): DetectMultivariateAnomaly = {
+    logFit({
+      import MADJsonProtocol._
+
+      val df = getInternalTransformer(dataset.schema)
+        .transform(dataset)
+        .withColumn("diagnosticsInfo", col(getOutputCol)
+          .getField("modelInfo").getField("diagnosticsInfo"))
+        .withColumn("modelId", col(getOutputCol).getField("modelId"))
+        .select(getOutputCol, "modelId", "diagnosticsInfo")
+        .collect()
+      this.setDiagnosticsInfo(df.head.get(2).asInstanceOf[GenericRowWithSchema]
+        .json.parseJson.convertTo[DiagnosticsInfo])
+      val modelId = df.head.getString(1)
+      new DetectMultivariateAnomaly()
+        .setSubscriptionKey(getSubscriptionKey)
+        .setLocation(getUrl.split("/".toCharArray)(2).split(".".toCharArray).head)
+        .setModelId(modelId)
+    })
+  }
+
+  override def copy(extra: ParamMap): Estimator[DetectMultivariateAnomaly] = defaultCopy(extra)
+
+  override def transformSchema(schema: StructType): StructType = {
+    getInternalTransformer(schema).transformSchema(schema)
```

Comment:
this wont yield the correct final schema because the pipeline returned doesn't include the DetectMultivariateAnomaly stuff. One option is to just implement all the code in fit, and manually implement the transform schema. Another is to put the  fit code into getInternalTransformer and then no need to override this.